src/binary-reader-linker.cc

```diff
@@ -125,7 +124,8 @@ static Result begin_section(BinaryReaderContext* ctx,
                             uint32_t size) {
   Context* context = static_cast<Context*>(ctx->user_data);
   LinkerInputBinary* binary = context->binary;
-  Section* sec = append_section(&binary->sections);
+  Section* sec = new Section();
+  binary->sections.emplace_back(sec);
```

Comment:
This case is a bit weird since sections is a vector of std::unique_ptr, so it is creating a new object, just not a Section. :-)